src/ast.c

```diff
@@ -267,6 +268,7 @@ static jl_ast_context_t *jl_ast_ctx_enter(void)
 
 static void jl_ast_ctx_leave(jl_ast_context_t *ctx)
 {
+    JL_SIGATOMIC_END();
```

Comment:
Fixed in https://github.com/JuliaLang/julia/commit/417e49aae87222d96df6125b188686da39a9bc80